### PR TITLE
Handle Escape Click for Zoomed images

### PR DESCRIPTION
**Pull request by dnlnvl**
This pull request closes (past link to GitHub issue here).

## Pre-Flight Checklist
I have made sure to
- [x] make sure this PR is set to merge to the correct `X-0-stable` branch
- [ ] add a Storybook example (if necessary)
- [x] manually test all the Storybook examples
  * run `$ yarn run storybook`
  * navigate to http://localhost:6006
- [x] run `$ yarn run build` to build the project code & static example
- [x] check that the example in `example/build/index.html` works: `$ open example/build/index.html`

### CODE REVIEW

File: source/Activated.js [hunk 1]
```diff
@@ -35,6 +35,13 @@ const Activated = ({
     setIsUnloading(true)
   }, [])
 
+  const handleEscClick = useCallback(e => {
+    e.preventDefault()
+    if (e.keyCode === 27) {
```
Comment (rpearce) [inline source/Activated.js]: As `keyCode` is deprecated on MDN (https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode), we should probably check for `e.key === 'Escape'` first!
```js
if (e.key === 'Escape' || e.keyCode === 27) {
```

File: source/Activated.js [hunk 2]
```diff
@@ -50,14 +57,17 @@ const Activated = ({
   // ========================================
   // = TELL PARENT THAT WE'RE ALL DONE HERE =
   // ========================================
+
   useEffect(() => {
     // @TODO: sync with transition duration?
     const unloadTimeout = isUnloading ? setTimeout(onDeactivate, 300) : null
 
+    document.addEventListener('keydown', handleEscClick, false)
```
Comment (rpearce) [inline source/Activated.js]: In the same area where we `clearTimeout(unloadTimeout)`, we also need to clean up this event listener:
```js
document.removeEventListener('keydown', handleEscClick, false)
```

File: source/Activated.js [hunk 3]
```diff
@@ -35,6 +35,13 @@ const Activated = ({
     setIsUnloading(true)
   }, [])
 
+  const handleEscClick = useCallback(e => {
```
Comment (rpearce) [inline source/Activated.js]: As you don't really "click" escape, we should probably name this `handleKeydown`, as `keydown` is the event we're listening for

File: source/Activated.js [hunk 4]
```diff
@@ -35,6 +35,13 @@ const Activated = ({
...
+    e.preventDefault()
```
Comment (rpearce) [inline source/Activated.js]: I think `e.preventDefault()` should only be triggered if it's the escape key, don't you think? I reckon this should live inside the `if (...) { ... }`

**Comment by dnlnvl** [comment]
Thanks for all the comments! They're very informative in my case, only been a dev for about a year now, and I started with ReactJS. I applied the changes based on your comments in another commit. I've enjoyed working on your project. It was really easy to read, very clean code. 👌

File: source/Activated.js (hunk 4 above)
Comment (dnlnvl) [reply source/Activated.js]: Applied!

File: source/Activated.js (hunk 3 above)
Comment (dnlnvl) [reply source/Activated.js]: Applied!

File: source/Activated.js (hunk 2 above)
Comment (dnlnvl) [reply source/Activated.js]: Applied!

File: source/Activated.js (hunk 1 above)
Comment (dnlnvl) [reply source/Activated.js]: Applied!

**Pull request merged**